### PR TITLE
[VMD] Format, unformat and transform text

### PR DESCRIPTION
**Pull request by philippelamy1**
## Description

We add an `unformatText` and `transformText` function and we fix the bindings for `formatText` (iOS)

## Motivation and Context

In our project we need to format a textfield value into phone number. We use `formatText` to format the value, `unformatText` to make sure only digits are passed to common, and `transformText` to limit the number of characters. The formatText in VMDTextfield (swift) wasn't working and we found out that since iOS 15 "it might be impossible to manipulate the binding itself" (source: https://itnext.io/swiftui-limit-the-amount-of-characters-in-a-textfield-view-906e819529a5)

So we fix this by replacing `text` by a `@State var` and we use the function `.onChange` to apply the transformations.

## How Has This Been Tested?

Directly into our project:

https://user-images.githubusercontent.com/71347410/174296366-5305598e-650f-4d28-a84b-d50841958f05.mp4

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: trikot-kword/sample/common/TRIKOT_FRAMEWORK_NAME.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |spec|
     spec.name                     = 'TRIKOT_FRAMEWORK_NAME'
-    spec.version                  = '3.3.4'
+    spec.version                  = '3.3.5'
```
Comment (remi) [inline trikot-kword/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:3]: I don’t think the version should be bumped _inside_ the feature PR.

Comment (hugolefrancois) [inline trikot-kword/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:3]: I think that this number will be incremented automatically by the release script? @npresseault

File: trikot-viewmodels-declarative/sample/common/TRIKOT_FRAMEWORK_NAME.podspec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |spec|
     spec.name                     = 'TRIKOT_FRAMEWORK_NAME'
-    spec.version                  = '3.3.4'
+    spec.version                  = '3.3.5'
```
Comment (hugolefrancois) [inline trikot-viewmodels-declarative/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:3]: I think that this number will be incremented automatically by the release script? @npresseault

**Review by hugolefrancois — APPROVED** [review]

File: trikot-kword/sample/common/TRIKOT_FRAMEWORK_NAME.podspec (hunk 1 above)
Comment (npresseault) [reply trikot-kword/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:3]: It is, the file is automatically generated. I wonder if we should keep it under source control.

File: trikot-viewmodels-declarative/swift/swiftui/Components/VMDTextField.swift [hunk 3]
```diff
@@ -12,13 +12,7 @@ public struct VMDTextField<Label>: View where Label: View {
...
     }
 
-    private var text: Binding<String> {
-        Binding {
-            self.viewModel.text
-        } set: { text in
-            self.viewModel.onValueChange(text: self.viewModel.formatText(text))
-        }
-    }
+    @State private var text: String = ""
```
Comment (antoinelamy) [inline trikot-viewmodels-declarative/swift/swiftui/Components/VMDTextField.swift:15]: Really not sure about that change, any text transformation should be handled by the view model anyway.

**Pull request merged**